roles/confluent.kafka_connect_replicator/tasks/health_check.yml

```diff
@@ -0,0 +1,46 @@
+---
+# Cannot use Block/Rescue in Ansible Handlers: https://github.com/ansible/ansible/issues/14270
+# Implementing try/catch logic with ignore_errors and conditionals
+- name: Wait for API to return 200
+  uri:
+    url: "{{kafka_connect_replicator_http_protocol}}://{{inventory_hostname}}:{{kafka_connect_replicator_port}}/connectors"
+    status_code: 200
+    validate_certs: false
+    # Basic auth header is redundant for non rbac install
+    url_username: "{{kafka_connect_replicator_health_check_user}}"
+    url_password: "{{kafka_connect_replicator_health_check_password}}"
+    force_basic_auth: true
+  register: result1
+  until: result1.status == 200
+  retries: 30
+  delay: 10
+  when: not kafka_connect_replicator_listener ['ssl_mutual_auth_enabled'] | default(ssl_mutual_auth_enabled) | bool
```

Comment:
There should be a variable that represents if the replicator has mtls enabled, its probably named `kafka_connect_replicator_ssl_mutual_auth_enabled` - that is the pattern used for other components.

`kafka_connect_replicator_listener ['ssl_mutual_auth_enabled']` represents if the listener replicator connects to kafka over has mtls or not